hydroflow/tests/surface_persist.rs

```diff
@@ -0,0 +1,75 @@
+use multiplatform_test::multiplatform_test;
+
+use hydroflow::util::collect_ready;
+use hydroflow::{assert_graphvis_snapshots, hydroflow_syntax};
+
+#[multiplatform_test]
+pub fn test_persist_basic() {
+    let (result_send, mut result_recv) = hydroflow::util::unbounded_channel::<u32>();
+
+    let mut hf = hydroflow_syntax! {
+        repeat_iter([1])
```

Comment:
surprised by the use of `repeat_iter` before `persist`. Seems like an anti-pattern.

Reply:
useful in tests, but I'll update the doc example to not use `repeat_iter`